File: TypeScript/arrays-and-hashing/two-sum/twoSumBest.ts
class SolutionBest {

    twoSum(nums: number[], target: number) : number[] {
        const vistos: Record<number,number> = {};
        for(let i=0; i<nums.length;i++){
            const complemento = target - nums[i];
            if(complemento in vistos){
                return [vistos[complemento],i]
            }
            vistos[nums[i]] = i;
        }
        return [];
    }

}

const numsB: number[] = [2,7,11,15]; 
const indexesB:number[] = new SolutionBest().twoSum(numsB, 9);
console.log(indexesB);

File: Python/arrays-and-hashing/two-sum/two_sum_best.py
from typing import List

class Solution():
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        vistos = {}
        for i, num, in enumerate(nums):
            complemento = target - num;
            if complemento in vistos:
                return [vistos[complemento],i]
            vistos[num] = i;
    
nums = [2,2,4,15]
indexes = Solution().twoSum(nums,4)
print(indexes)